lang/cs/Org.Apache.REEF.Wake/Remote/IType.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Interface that classes should implement to express their type
+    /// </summary>
+    public interface IType
+    {
+        /// <summary>
+        /// Returns the type of the class
+        /// </summary>
+        Type ClassType { get; set; }
```

Comment:
OK. Why not use `.GetType()` as suggested in the [typeof documentation](https://msdn.microsoft.com/en-us/library/58918ffs.aspx)?